File: DynamoDB-SDK-Examples/python/WorkingWithTables/disable_auto-scaling.py
import boto3
aas_client = boto3.client('application-autoscaling')
#iam_client = boto3.client('iam')

table_name = "Music" # Add the name of the DynamoDB table you want to add auto-scaling to between the double quotes.

# detach the Write scaling policy to the table.
response = aas_client.delete_scaling_policy(
    PolicyName='{}ScalingPolicy'.format(table_name),
    ServiceNamespace='dynamodb',
    ResourceId='table/{}'.format(table_name),
    ScalableDimension='dynamodb:table:WriteCapacityUnits'
)

# detach the Read scaling policy to the table.
response = aas_client.delete_scaling_policy(
    PolicyName='{}ScalingPolicy'.format(table_name),
    ServiceNamespace='dynamodb',
    ResourceId='table/{}'.format(table_name),
    ScalableDimension='dynamodb:table:ReadCapacityUnits'
)

# Deregister the RCU targets for the table
response = aas_client.deregister_scalable_target(
    ServiceNamespace='dynamodb',
    ResourceId='table/{}'.format(table_name),
    ScalableDimension='dynamodb:table:ReadCapacityUnits',
)

# Deregister the WCU targets for the table
response = aas_client.deregister_scalable_target(
    ServiceNamespace='dynamodb',
    ResourceId='table/{}'.format(table_name),
    ScalableDimension='dynamodb:table:WriteCapacityUnits',
)

File: examples/SDK/node.js/sdk_v3/control_plane/WorkingWithTables/disable_auto_scaling.js
const { 
  ApplicationAutoScalingClient, 
  DeleteScalingPolicyCommand, 
  DeregisterScalableTargetCommand 
} = require("@aws-sdk/client-application-auto-scaling");

const applicationAutoscaling = new ApplicationAutoScalingClient({
  region: "us-west-2",
});

const tableName = "Music"; // Add the name of the DynamoDB table you want to add auto-scaling to between the double quotes.

const disableAutoScaling = async () => {
  // Delete the Read scaling policy from the table.
  let command = new DeleteScalingPolicyCommand({
    PolicyName: `${tableName}ScalingPolicy`,
    ServiceNamespace: "dynamodb",
    ResourceId: `table/${tableName}`,
    ScalableDimension: "dynamodb:table:ReadCapacityUnits",
  });
  let response = await applicationAutoscaling.send(command);

  command = new DeleteScalingPolicyCommand({
    PolicyName: `${tableName}ScalingPolicy`,
    ServiceNamespace: "dynamodb",
    ResourceId: `table/${tableName}`,
    ScalableDimension: "dynamodb:table:WriteCapacityUnits",
  });
  response = await applicationAutoscaling.send(command);

  // Deregister the RCU targets for the table
  command = new DeregisterScalableTargetCommand({
    ServiceNamespace: "dynamodb",
    ResourceId: `table/${tableName}`,
    ScalableDimension: "dynamodb:table:ReadCapacityUnits",
  });
  response = await applicationAutoscaling.send(command);

  // Deregister the WCU targets for the table
  command = new DeregisterScalableTargetCommand({
    ServiceNamespace: "dynamodb",
    ResourceId: `table/${tableName}`,
    ScalableDimension: "dynamodb:table:WriteCapacityUnits",
  });
  response = await applicationAutoscaling.send(command);

  console.log("Autoscaling has been disabled");
};

disableAutoScaling().catch((error) => console.error(JSON.stringify(error, null, 2)));
